### PR TITLE
data/ipfs: upgrade IPFS CID to v1

### PR DESCRIPTION
**Pull request by altergui**
data/ipfs: upgrade IPFS CID to v1 

 * reimplement Publish using i.CoreApi
 * move unixfsFilesNode out of AddAndPin
 * reimplement AddAndPin using i.CoreAPI
 * lint Pin and Unpin
 * simpler PublishIPNSpath, refactoring addAndPin to return corepath.Resolved
 * use IPFSCIDv1, drop IPFScontentIdentifier that returned v0
 * new func MockStorage needed by api/census_test

### CODE REVIEW

**Comment by altergui** [comment]
(base is wrong, i need to rebase on master, solving conflicts)

File: data/ipfs.go [hunk 1]
```diff
@@ -303,13 +295,12 @@ func (i *IPFSHandle) Retrieve(ctx context.Context, path string, maxSize int64) (
...
 // so the caller must handle properly the logic by using goroutines, channels or other
 // mechanisms in order to not block the whole program execution.
-func (i *IPFSHandle) PublishIPNSpath(ctx context.Context, path string,
-	keyalias string) (coreiface.IpnsEntry, error) {
-	root, err := i.AddAndPin(ctx, path)
-	if err != nil {
-		return nil, err
-	}
-	c, err := cid.Parse(root)
+func (i *IPFSHandle) PublishIPNSpath(
```
Comment (p4u) [inline data/ipfs.go]: I know some linters want to split the parameters this way, but we decided not to follow the approach. Instead we split the lines in the minimum number while trying not to surpass 111 chars and make them as symmetric as possible.

**Review by p4u — CHANGES_REQUESTED** [review]
@altergui there were some commits on the original branch (the one I started) that I'd like to maintain. Such as the ones modifying ipfsSync to allow "onlyConnect" mode.

**Comment by p4u** [comment]
> 

Ah sorry, I miss read the branch name. So my changes are here https://github.com/vocdoni/vocdoni-node/tree/f/ipfs_sync_fixes
Could you cherry-pick those that do not collide with your work?

File: data/ipfs.go (hunk 1 above)
Comment (altergui) [reply data/ipfs.go]: good to know, i dropped that gofumptd commit

**Comment by altergui** [comment]
> > 
> 
> Ah sorry, I miss read the branch name. So my changes are here https://github.com/vocdoni/vocdoni-node/tree/f/ipfs_sync_fixes Could you cherry-pick those that do not collide with your work?

ok but they seem out of scope for this PR so i'll prepare a separate one, review this as-is

**Review by p4u — APPROVED** [review]
The changes LGTM, the whole implementation can probably be improved but that's nothing we can do now. @mvdan can you also take a look?

**Review by mvdan — APPROVED** [review]
SGTM

**Comment by altergui** [comment]
squashed everything and added a simple unit test FTW

**Pull request merged**